scripts/generate-meetup-data.js

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env node
+var _ = require('lodash');
+var fs = require('fs');
+var targetFileName = 'free-code-camp-events.mock.json';
+var targetPath = 'js/';
+var meetup_api_key = _.get(process, 'env.MEETUP_API_KEY');
+var targetGroupUrlname = 'Free-Code-Camp-Indy';
+var targetStatus = ['upcoming', 'past'];
+
+function generateNewMeetupMockData(meetupApiKey) {
+  var meetup = require('meetup-api')({key: meetupApiKey});
+  return new Promise(function (resolve, reject) {
+    meetup.getEvents({group_urlname: targetGroupUrlname, status: targetStatus}, function (err, response) {
+      if (err) {
+        reject(new Error(err));
+      } else {
+        // Strip property that has secrets information!
+        var results = {
+          results: _.get(response, 'results')
+        }
+        resolve(results);
+      }
+    });
+  });
+};
+function writeNewData(mockJson) {
+  return new Promise(function (resolve, reject) {
+    fs.writeFile(targetPath + targetFileName, mockJson, function (err) {
+      if (err) {
+        reject(new Error(err))
+      } else {
+        resolve('Successfully written to file!');
+      }
+    });
+  });
+};
+function handleError(error) {
+  console.error(error);
+  process.exit(1);
+}
+
+// "main"
+if (_.isUndefined(meetup_api_key)) {
+  var error = new Error('MEETUP_API_KEY environment variable not found.  Please reference the README.md for instructions on how to set this.  Ending script.');
+  handleError(error);
+} else {
+  // TODO: Refactor this "callback arrow" antipattern to perhaps use ESNext's `async`/`await` functionality.
+  var getMockJson = generateNewMeetupMockData(meetup_api_key);
+  getMockJson.then(function (mockJson) {
+    var formattedMockJson = JSON.stringify(mockJson, null, 2);
+    var writeResult = writeNewData(formattedMockJson);
+    writeResult.then(function () {
+      console.log('Successfully wrote new data to target file');
+      process.exit(0);
+    }, handleError);
+  }, handleError);
+}
```

Comment:
For the group we have working on this, I'd prefer to keep away from over-engineering solutions. There's a reason why this simple static site isn't built using a framework. As a group, we decided that it would be too much of a learning curve, for very little benefit, and not really necessary for what we are trying to do.

I'd prefer our Node code be written in a style that will be compatible with as many versions of Node as possible. Though we have no set version we are trying to maintain compatibility with, we shouldn't be intentionally forcing people to upgrade (in case they have to use a specific version for work), or learn new tools (nvm/babel/etc), just to be able to use newer syntactic sugar. This logic seems to fall in line with the groups decision for technology usage on the site (learning curve/benefit/functional necessity).